### DISCUSSION TITLE
How do you explicilty set a total table width?

### DISCUSSION THREAD

**Discussion by Eutropios** [typst/typst#3486] [Q&A] [answered]
I wanted to create a header for a worksheet, but I ran into a roadblock with the total table width. More specifically, the problem is that I cannot set a width for the table to make it stretch to the width of the page.
```typst
#align(center)[
    #table(
        columns: 3,
        align: (col, row) => (left, center, right).at(col),
        inset: 2pt,
        stroke: none,
        [#strong[]],
        [],
        [Noah Jenner],
        [#strong[Course: ]],
        [],
        [100336212],
        [#strong[Instructor: ]],
        [],
        [MM/DD/YYYY],
    )
]
```
<img width="426" alt="WindowsTerminal_Cdo2zvrAqO" src="https://github.com/typst/typst/assets/64815328/556f967d-a01e-4d6d-bea9-def75ce54ac2">

**Comment by xkevio** [comment, marked as answer]
The `columns` parameter can be used to specify the lengths of each column as well in the form of an array.
In order to make it span the whole page, you can use `1fr` for each column's length as it means "take up all of the remaining space".

So you can do: `columns: (1fr, 1fr, 1fr)` meaning a table with three columns taking up as much space as possible each.

**Reply by Eutropios** [reply]
When would it be wise to choose each of the following solutions, then?

```typst
#align(center)[
    #table(
        columns: (4fr, 5fr, 4fr),
        align: (col, row) => (left, center, right).at(col),
        inset: 2pt,
        stroke: none,
        [#strong[Assignment \#6]],
        [],
        [Noah Jenner],
        [#strong[class name]],
        [],
        [foobar],
        [#strong[Instructor:] foobar],
        [],
        [MM/DD/YYYY],
    )
]

// versus

#grid(
    columns: (4fr, 5fr, 4fr),
    rows: (auto),
    row-gutter: 5pt,
    strong[Assignment \#6],
    block[],
    align(right)[Noah Jenner],
    strong[class name],
    block[],
    align(right)[\#\#\#\#],
    block[#strong[Instructor:] foobar],
    block[],
    align(right)[MM/DD/YYYY]
)
```

**Reply by xkevio** [reply]
Semantically, a grid is probably closer to what you are creating. Since by default a table also has an inset and a stroke, a grid might be easier to use too besides manually specifying alignment for each column. In the next update, the underlying APIs between grid and table will be unified, so it won't matter as much.

**Reply by Eutropios** [reply]
Ahhh I see, that makes a lot of sense. Thank you for the help!
